extensions/iceberg/src/main/java/io/deephaven/iceberg/util/IcebergCatalogAdapter.java

```diff
@@ -249,8 +256,124 @@ public IcebergTableAdapter loadTable(@NotNull final TableIdentifier tableIdentif
     /**
      * Returns the underlying Iceberg {@link Catalog catalog} used by this adapter.
      */
-    @SuppressWarnings("unused")
     public Catalog catalog() {
         return catalog;
     }
+
+    /**
+     * Create a new Iceberg table in the catalog with the given table identifier and definition.
+     * <p>
+     * All columns of type {@link ColumnDefinition.ColumnType#Partitioning partitioning} will be used to create the
+     * partition spec for the table.
+     *
+     * @param tableIdentifier The identifier of the new table.
+     * @param definition The {@link TableDefinition} of the new table.
+     * @return The {@link IcebergTableAdapter table adapter} for the new Iceberg table.
+     */
+    public IcebergTableAdapter createTable(
+            @NotNull final TableIdentifier tableIdentifier,
+            @NotNull final TableDefinition definition) {
+        // TODO Add these APIs to python code once finalized
+        final IcebergUtils.SpecAndSchema specAndSchema = IcebergUtils.createSpecAndSchema(
+                definition, IcebergParquetWriteInstructions.DEFAULT);
+        return createTable(tableIdentifier, specAndSchema.schema(), specAndSchema.partitionSpec());
+    }
+
+    /**
+     * Create a new Iceberg table in the catalog with the given table identifier, schema, and partition spec.
+     *
+     * @param tableIdentifier The identifier of the new table.
+     * @param schema The schema of the new table.
+     * @param partitionSpec The partition spec of the new table.
+     * @return The {@link IcebergTableAdapter table adapter} for the new Iceberg table.
+     */
+    private IcebergTableAdapter createTable(
+            @NotNull final TableIdentifier tableIdentifier,
+            @NotNull Schema schema,
+            @NotNull PartitionSpec partitionSpec) {
+        final boolean newNamespaceCreated = createNamespaceIfNotExists(tableIdentifier.namespace());
+        try {
+            final org.apache.iceberg.Table table =
+                    catalog.createTable(tableIdentifier, schema, partitionSpec,
+                            Map.of(TableProperties.DEFAULT_FILE_FORMAT, TableProperties.DEFAULT_FILE_FORMAT_DEFAULT));
+            return new IcebergTableAdapter(tableIdentifier, table, dataInstructionsProvider);
+        } catch (final Throwable throwable) {
+            if (newNamespaceCreated) {
+                // Delete it to avoid leaving a partial namespace in the catalog
+                try {
+                    dropNamespaceIfExists(tableIdentifier.namespace());
+                } catch (final RuntimeException dropException) {
+                    throwable.addSuppressed(dropException);
+                }
+            }
+            throw throwable;
+        }
+    }
+
+    public IcebergTableAdapter createTableAndAppend(
```

Comment:
Why have this method if we're not doing a transactional create + append?

Reply:
Adding a transactional create + append would actually be a bigger change than I was expecting, since currently, IcebergTableAdapter class assumes that the table exists. So I would need to make much bigger changes such that I can create an adapter without a table and then create the table and write to it in a single transaction.
I am leaving it out of the scope for now, but will add this to a ticket for future work.